Updated Caffe implementation to be modular

PalashTanejaPro opened the pull request:
IMPORTANT NOTES (please read and then delete):

* The PR title should start with "Fix #bugnum: " (if applicable), followed by a clear one-line present-tense summary of the changes introduced in the PR. For example: "Fix #bugnum: Changed UI layout from horizontal to vertical.".

* Please make sure to mention "#bugnum" somewhere in the description of the PR. This enables Github to link the PR to the corresponding bug.

Please also make sure to follow the [contribution](https://github.com/Cloud-CV/Fabrik/blob/master/.github/CONTRIBUTING.md) guidelines.

virajprabhu commented:
Thanks @PalashTanejaPro, will review. Can you please add a small description of how you've implemented the PR?

PalashTanejaPro commented:
@virajprabhu I was asked in the GCI task to convert the current caffe implementation to be modular just like the keras one. So, I check the caffe_app directory and converted the file with the most amount of if-else statements into a modular implementation. 

To do that, I created a function for each layer type. I then created a dictionary that relates the layer.type with the layer's function. Finally, I removed all the if-else statement and replaced it with just one function call ```layer_dict[layer.type](layer)```. This calls the corresponding function for the layer.
